Source/DataSources/I3SDataSource.js

```diff
@@ -0,0 +1,4601 @@
+    });
+
+    // create a new mesh for this page
+    meshes.push({
+      primitives: [
+        {
+          attributes: attributes,
+          indices: indicesIndex,
+          material: 0,
+        },
+      ],
+    });
+    nodesInScene.push(0);
+    nodes.push({ mesh: 0 });
+
+    return {
+      buffers: buffers,
+      bufferViews: bufferViews,
+      accessors: accessors,
+      meshes: meshes,
+      nodes: nodes,
+      nodesInScene: nodesInScene,
+    };
+  }
+
+  /*
+  ..######...########..#######..##.....##.########.########.########..##....##
+  .##....##..##.......##.....##.###...###.##..........##....##.....##..##..##.
+  .##........##.......##.....##.####.####.##..........##....##.....##...####..
+  .##...####.######...##.....##.##.###.##.######......##....########.....##...
+  .##....##..##.......##.....##.##.....##.##..........##....##...##......##...
+  .##....##..##.......##.....##.##.....##.##..........##....##....##.....##...
+  ..######...########..#######..##.....##.########....##....##.....##....##...
+
+  .########..########..######...#######..########..########.########.
+  .##.....##.##.......##....##.##.....##.##.....##.##.......##.....##
+  .##.....##.##.......##.......##.....##.##.....##.##.......##.....##
+  .##.....##.######...##.......##.....##.##.....##.######...########.
+  .##.....##.##.......##.......##.....##.##.....##.##.......##...##..
+  .##.....##.##.......##....##.##.....##.##.....##.##.......##....##.
+  .########..########..######...#######..########..########.##.....##
+  */
+
+  function decode(data, schema, bufferInfo, featureData) {
+    var magicNumber = new Uint8Array(data, 0, 5);
+    if (
+      magicNumber[0] === "D".charCodeAt() &&
+      magicNumber[1] === "R".charCodeAt() &&
+      magicNumber[2] === "A".charCodeAt() &&
+      magicNumber[3] === "C".charCodeAt() &&
+      magicNumber[4] === "O".charCodeAt()
+    ) {
+      return decodeDracoEncodedGeometry(data, bufferInfo);
+    }
+    return decodeBinaryGeometry(data, schema, bufferInfo, featureData);
+  }
+
+  /*
+  .########..########.....###.....######...#######.
+  .##.....##.##.....##...##.##...##....##.##.....##
+  .##.....##.##.....##..##...##..##.......##.....##
+  .##.....##.########..##.....##.##.......##.....##
+  .##.....##.##...##...#########.##.......##.....##
+  .##.....##.##....##..##.....##.##....##.##.....##
+  .########..##.....##.##.....##..######...#######.
+
+  .########..########..######...#######..########..########.########.
+  .##.....##.##.......##....##.##.....##.##.....##.##.......##.....##
+  .##.....##.##.......##.......##.....##.##.....##.##.......##.....##
+  .##.....##.######...##.......##.....##.##.....##.######...########.
+  .##.....##.##.......##.......##.....##.##.....##.##.......##...##..
+  .##.....##.##.......##....##.##.....##.##.....##.##.......##....##.
+  .########..########..######...#######..########..########.##.....##
+  */
+
+  function decodeDracoEncodedGeometry(data, bufferInfo) {
```

Comment:
The functionality of [`DracoLoader`](https://github.com/CesiumGS/cesium/blob/main/Source/Scene/DracoLoader.js) can be extended with these functions here. Even though it currently may be very glTF-specific, it should be ok to add new functions there that simply transform the data.